src/main/java/org/terasology/projsndwv/genetics/Genome.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.projsndwv.genetics;
+
+import org.terasology.projsndwv.genetics.components.GeneticsComponent;
+import org.terasology.utilities.random.MersenneRandom;
+
+import java.util.*;
+
+public class Genome {
+    private MersenneRandom random;
+    private final int size;
+
+    private List<Map<AllelePair, List<Mutation>>> mutations;
+
+    public Genome(int size, long seed) {
+        this.size = size;
+        random = new MersenneRandom(seed);
+
+        mutations = new ArrayList<>(size);
+
+        for (int i = 0; i < size; i++) {
+            mutations.add(new HashMap<>());
+        }
+    }
+
+    public Iterator<GeneticsComponent> combine(GeneticsComponent g0, GeneticsComponent g1) {
+        if (!g0.isValid() || !g1.isValid()) {
+            return null;
+        }
+        else if (g0.size != size || g1.size != size) {
+            return null;
+        }
+
+        List<Mutation> possibleMutations = new ArrayList<>();
+
+        for (int i = 0; i < size; i++) {
+            Map<AllelePair, List<Mutation>> mutationsForLocus = mutations.get(i);
+            AllelePair ap = new AllelePair(g0.activeGenes.get(i), g1.activeGenes.get(i)); // TODO: Remove dependence on active genes
+            if (mutationsForLocus.containsKey(ap)) {
+                possibleMutations.addAll(mutationsForLocus.get(ap));
+            }
+        }
+
+        return new Iterator<GeneticsComponent>() {
+            @Override
+            public boolean hasNext() {
+                return true;
+            }
+
+            @Override
+            public GeneticsComponent next() {
+                GeneticsComponent result = new GeneticsComponent(size);
+
+                GeneticsComponent g0mutated = g0;
+                GeneticsComponent g1mutated = g1;
+
+                Collections.shuffle(possibleMutations);
```

Comment:
Assuming the purpose of both this shuffle and the next shuffle is to try and reduce the chance that the same mutation is picked for each one?

Either way it would be nice to see it perhaps split into a separate method (with appropriate javadoc ofc)

Reply:
More so that without the shuffles mutations that are added to the genome later are less likely. With the shuffles it should be completely random and converge to the base chance.

Really obvious point for a method missed, oops. Will do.